### PR TITLE
[BUG]: Visualizations not appearing in Operational Panels fixed

### PR DESCRIPTION
**Pull request by SivaprasadAluri**
### Description
Saved visualizations will appear in operational panel after creating the panel

### Issues Resolved
[BUG]: Visualizations not appearing in Operational Panels (https://github.com/opensearch-project/observability/issues/907)

### CODE REVIEW

File: dashboards-observability/public/components/custom_panels/helpers/utils.tsx [hunk 1]
```diff
@@ -299,7 +299,7 @@ export const displayVisualization = (metaData: any, data: any, type: string) =>
         rawVizData: data,
         query: { rawQuery: metaData.query },
         indexFields: {},
-        userConfigs: data?.jsonData,
+        userConfigs: {},
```
Comment (mengweieric) [inline dashboards-observability/public/components/custom_panels/helpers/utils.tsx:302]: Why are we change it from metaData.user_configs to data?.jsonData and removed it in the end? Did we do BWC?

Comment (ps48) [reply dashboards-observability/public/components/custom_panels/helpers/utils.tsx:302]: +1 @SivaprasadAluri Can you please explain why was this change added?

File: dashboards-observability/public/components/custom_panels/helpers/utils.tsx [hunk 2]
```diff
@@ -297,9 +297,9 @@ export const displayVisualization = (metaData: any, data: any, type: string) =>
       visualizations={getVizContainerProps({
         vizId: type,
         rawVizData: data,
-        query: {},
+        query: { rawQuery: metaData.query === undefined ? {} : metaData.query },
```
Comment (ps48) [inline dashboards-observability/public/components/custom_panels/helpers/utils.tsx:300]: In query parameter Why are we checking `metaData.query === undefined`? The statement will never be reached while being true. Check: https://github.com/opensearch-project/observability/blob/d243e918aff872efe7fed3769662ebddfec56b2e/dashboards-observability/public/components/custom_panels/helpers/utils.tsx#L292-L294

You may simply change this to:
```
query: { rawQuery: metaData.query }
```

File: dashboards-observability/test/panels_constants.tsx [hunk 3]
```diff
@@ -12,10 +12,23 @@ export const sampleSavedVisualization = {
   visualization: {
     id: 'oiuccXwBYVazWqOO1e06',
     name: 'Flight Count by Origin',
-    query:
-      'source=opensearch_dashboards_sample_data_flights | fields Carrier,FlightDelayMin | stats sum(FlightDelayMin) as delays by Carrier',
+    query: {
+      rawQuery:
```
Comment (ps48) [inline dashboards-observability/test/panels_constants.tsx:16]: @SivaprasadAluri @mengweieric I don't see this structure change in the backend: https://github.com/opensearch-project/observability/blob/main/opensearch-observability/src/main/kotlin/org/opensearch/observability/model/SavedVisualization.kt#L24-L25

Are we yet to have a change in the backend for this?

File: dashboards-observability/public/components/custom_panels/helpers/utils.tsx (hunk 1 above)
Comment (SivaprasadAluri) [reply dashboards-observability/public/components/custom_panels/helpers/utils.tsx]: @mengweieric,@ps48 sorry i had reverted back to metaData.user_configs but missed the change during the last commit, its been reverted and tested.

File: dashboards-observability/public/components/custom_panels/helpers/utils.tsx (hunk 2 above)
Comment (SivaprasadAluri) [reply dashboards-observability/public/components/custom_panels/helpers/utils.tsx:300]: @ps48 we do have a test case where, we are checking with empty visualization, so i had to fix it with this. https://github.com/opensearch-project/observability/blob/e4a199be779eac368d0682fc636ee8d12cbaa05b/dashboards-observability/public/components/custom_panels/helpers/__tests__/utils.test.tsx#L128

Please let me know if we have to remove the test case or update the test case with valid visualization data.

File: dashboards-observability/test/panels_constants.tsx (hunk 3 above)
Comment (SivaprasadAluri) [reply dashboards-observability/test/panels_constants.tsx]: @ps48 rawQuery is not needed here, and its been reverted back to query

**Comment by mengweieric** [comment]
This PR is pending BWC support/updates as discussed with persist team.

**Comment by SivaprasadAluri** [comment]
The bug will be fixed in backward compatibility PR, this Pull is no longer needed.

**Pull request closed without merging**